index.js

```diff
@@ -0,0 +1,109 @@
+'use strict';
+
+var aws = require('aws-sdk-promise');
+var lodash = require('lodash');
+var debug = require('debug')('multi-region-promised-aws');
+var Promise = require('promise');
+
+// TODO: allow for each region to have its own defaults and settings
+function MultiAWS(nameOrConstructor, config, regions) {
+  var that = this;
+  this.apiObjs = {};
+
+  var constructor;
+  if (typeof nameOrConstructor === 'string') {
+    // They all use upper case... right?
+    if (Object.keys(aws).indexOf(nameOrConstructor) == -1) {
+      throw new Error(nameOrConstructor + ' is not a valid Aws Product');
+    }
+    constructor = aws[nameOrConstructor];
+    MultiAWS[nameOrConstructor] = aws[nameOrConstructor];
+  } else if (typeof nameOrConstructor === 'function') {
+    constructor = nameOrConstructor;
+  } else {
+    throw new Error('First argument must be a string or function');
+  }
+  
+  if (config.region) {
+    throw new Error('To avoid a footgun, your AWS config should not specify region');
+  }
+
+  // All I care that regions has a forEach, map and filter.  I don't
+  // care if it will return true for Array.isArray()
+  if (!regions.forEach || !regions.map || !regions.filter) {
```

Comment:
This is `node` I usually trusts `instanceof Array`, isArray is probably better this is also okay... Just saying...